tests/integration/integrationtests.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import sys
+import unittest
+from random import randint
+
+import utils
+
+
+class IntegrationTest(unittest.TestCase):
+    # helper funtion: test response time
+    def assert_response_time(self, elapsed_seconds, max_elapsed_seconds):
+        logging.debug(f"Request took {elapsed_seconds} second(s)")
+        self.assertLess(elapsed_seconds, max_elapsed_seconds)
+
+    def test_bad_request(self):
+        res = utils.get_by_params(None)
+        self.assert_response_time(res.elapsed.total_seconds(), 1)
+        self.assertEqual(res.status_code, 400)
+
+    def test_get_by_osu_id(self):
+        for _ in range(5):
+            random_osu_id = f"93{randint(0000000, 9999999)}"
+            with self.subTest(random_osu_id=random_osu_id):
+                res = utils.get_by_params({'osuId': osu_id})
+
+                self.assertIn('data', res.json())
+                self.assertIsInstance(res.json()['data'], list)
+                self.assert_response_time(res.elapsed.total_seconds(), 5)
+                self.assertEqual(res.status_code, 200)
+
+    def test_get_by_invalid_osu_id(self):
+        res = utils.get_by_params({'osuId': 'invalid_osu_id'})
+        res_data = res.json()['data']
+
+        self.assert_response_time(res.elapsed.total_seconds(), 1)
+        self.assertEqual(res.status_code, 200)
+        self.assertEqual(res_data, [])
+
+    def test_get_by_term_id(self):
+        for _ in range(3):
+            random_term_id = f"{randint(2000, 2018)}0{randint(0, 3)}"
```

Comment:
Could you add some logging to these tests? It would be helpful to log which random term ID was used. Same for the test that generates a random OSU ID - it would be helpful to log which ID was used for testing.